Domain.Sql/ReservedValue.cs

```diff
@@ -6,7 +6,20 @@
 namespace Microsoft.Its.Domain.Sql
 {
     public class ReservedValue
-    {
+    {
+        public ReservedValue()
+        {
+        }
+
+        public ReservedValue(ReservedValue other)
+        {

```

Comment:
I'm not sure this needs to be public or to be a constructor. This is more of a clone pattern, really (although `ICloneable` should be avoided since it returns `object`.)

From a maintenance perspective, the `MappingExpression` class can be interesting for cases like this because it's guaranteed to do the right thing even if fields are added / remove / renamed.